apps/scan/backend/src/scanners/custom/state_machine.ts

```diff
@@ -181,45 +181,62 @@ async function closeCustomClient({ client }: Context) {
     scannerStatus.sensorInputLeftLeft === SensorStatus.PaperPresent &&
     scannerStatus.sensorInputCenterLeft === SensorStatus.PaperPresent &&
     scannerStatus.sensorInputCenterRight === SensorStatus.PaperPresent &&
     scannerStatus.sensorInputRightRight === SensorStatus.PaperPresent;
-  const backHasPaper =
-    scannerStatus.sensorOutputLeftLeft === SensorStatus.PaperPresent &&
-    scannerStatus.sensorOutputCenterLeft === SensorStatus.PaperPresent &&
-    scannerStatus.sensorOutputCenterRight === SensorStatus.PaperPresent &&
+  const someFrontSensorsDetectPaper =
+    scannerStatus.sensorInputLeftLeft === SensorStatus.PaperPresent ||
+    scannerStatus.sensorInputCenterLeft === SensorStatus.PaperPresent ||
+    scannerStatus.sensorInputCenterRight === SensorStatus.PaperPresent ||
+    scannerStatus.sensorInputRightRight === SensorStatus.PaperPresent;
+
+  const someBackSensorsUnderRollersDetectPaper =
+    scannerStatus.sensorOutputCenterLeft === SensorStatus.PaperPresent ||
+    scannerStatus.sensorOutputCenterRight === SensorStatus.PaperPresent;
+  const someBackSensorsPastRollersDetectPaper =
+    scannerStatus.sensorOutputLeftLeft === SensorStatus.PaperPresent ||
     scannerStatus.sensorOutputRightRight === SensorStatus.PaperPresent;
+  const someBackSensorsDetectPaper =
+    someBackSensorsUnderRollersDetectPaper ||
+    someBackSensorsPastRollersDetectPaper;
 
   if (scannerStatus.isScannerCoverOpen) {
     return { type: 'SCANNER_DISCONNECTED' };
   }
 
   if (scannerStatus.isPaperJam || scannerStatus.isJamPaperHeldBack) {
-    if (!frontHasPaper && !backHasPaper) {
+    if (!someFrontSensorsDetectPaper && !someBackSensorsDetectPaper) {
       return { type: 'SCANNER_JAM_CLEARED' };
     }
 
     if (scannerStatus.isDoubleSheet) {
       return { type: 'SCANNER_JAM_DOUBLE_SHEET' };
     }
+
     return { type: 'SCANNER_JAM' };
   }
 
-  if (!frontHasPaper && !backHasPaper) {
+  // Paper past the back rollers but caught on, say, the emergency ballot bag should not be
+  // considered still in the scanner. Continuing to spin the rollers won't clear the paper. Rather,
+  // scanning the next ballot will.
+  if (!allFrontSensorsDetectPaper && !someBackSensorsUnderRollersDetectPaper) {
     return { type: 'SCANNER_NO_PAPER' };
   }
 
-  if (frontHasPaper && !backHasPaper) {
+  // Wait for all and not just some of the front sensors to detect paper before scanning to ensure
+  // proper alignment
+  if (allFrontSensorsDetectPaper && !someBackSensorsUnderRollersDetectPaper) {
     return { type: 'SCANNER_READY_TO_SCAN' };
   }
 
-  if (!frontHasPaper && backHasPaper) {
+  if (!someFrontSensorsDetectPaper && someBackSensorsUnderRollersDetectPaper) {
     return { type: 'SCANNER_READY_TO_EJECT' };
   }
 
-  // If we get here frontHasPaper and backHasPaper are true
+  // The above conditions make it such that this condition is equivalent to:
+  // someFrontSensorsDetectPaper && someBackSensorsUnderRollersDetectPaper
```

Comment:
This last case is the one that's relevant for this issue, since the first ballot (the one to be accepted) is at the back of the scanner and the second ballot is at the front. Previously, this condition was only hit if `allFrontSensorsDetectPaper && allBackSensorsDetectPaper`

Reply:
Might be worth an explicit check of the condition here followed by a `throwIllegalValue` or something like that to ensure we've covered all the cases

Reply:
Good call! Updated in https://github.com/votingworks/vxsuite/pull/4395/commits/f2dc1adb0e2eafd59e96bc838a85fe2ee053e6ae and added a fast-check test for extra confidence